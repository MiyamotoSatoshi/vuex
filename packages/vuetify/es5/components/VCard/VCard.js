'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; // Styles

// Extensions

// Mixins

// Helpers


require('../../../src/stylus/components/_cards.styl');

var _VPaper = require('../VPaper');

var _VPaper2 = _interopRequireDefault(_VPaper);

var _routable = require('../../mixins/routable');

var _routable2 = _interopRequireDefault(_routable);

var _mixins = require('../../util/mixins');

var _mixins2 = _interopRequireDefault(_mixins);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* @vue/component */
exports.default = (0, _mixins2.default)(_routable2.default, _VPaper2.default).extend({
    name: 'v-card',
    props: {
        elevation: {
            type: [Number, String],
            default: 2
        },
        flat: Boolean,
        hover: Boolean,
        img: String,
        raised: Boolean
    },
    computed: {
        classes: function classes() {
            return _extends({
                'v-card': true,
                'v-card--hover': this.hover
            }, _VPaper2.default.options.computed.classes.call(this));
        },
        computedElevation: function computedElevation() {
            if (this.flat) return 0;
            if (this.raised) return 3;
            return _VPaper2.default.options.computed.computedElevation.call(this);
        },
        styles: function styles() {
            var style = _extends({}, _VPaper2.default.options.computed.styles.call(this));
            if (this.img) {
                style.background = 'url("' + this.img + '") center center / cover no-repeat';
            }
            return style;
        }
    },
    render: function render(h) {
        var _generateRouteLink = this.generateRouteLink(this.classes),
            tag = _generateRouteLink.tag,
            data = _generateRouteLink.data;

        data.style = this.styles;
        return h(tag, this.setBackgroundColor(this.color, data), this.$slots.default);
    }
});
//# sourceMappingURL=VCard.js.map