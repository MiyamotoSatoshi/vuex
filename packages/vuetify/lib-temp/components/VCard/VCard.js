// Styles
import '../../stylus/components/_cards.styl';
// Extensions
import VPaper from '../VPaper';
// Mixins
import Routable from '../../mixins/routable';
// Helpers
import mixins from '../../util/mixins';
/* @vue/component */
export default mixins(Routable, VPaper).extend({
    name: 'v-card',
    props: {
        elevation: {
            type: [Number, String],
            default: 2
        },
        flat: Boolean,
        hover: Boolean,
        img: String,
        raised: Boolean
    },
    computed: {
        classes() {
            return {
                'v-card': true,
                'v-card--hover': this.hover,
                ...VPaper.options.computed.classes.call(this)
            };
        },
        computedElevation() {
            if (this.flat)
                return 0;
            if (this.raised)
                return 3;
            return VPaper.options.computed.computedElevation.call(this);
        },
        styles() {
            const style = {
                ...VPaper.options.computed.styles.call(this)
            };
            if (this.img) {
                style.background = `url("${this.img}") center center / cover no-repeat`;
            }
            return style;
        }
    },
    render(h) {
        const { tag, data } = this.generateRouteLink(this.classes);
        data.style = this.styles;
        return h(tag, this.setBackgroundColor(this.color, data), this.$slots.default);
    }
});
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiVkNhcmQuanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQ2FyZC9WQ2FyZC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxTQUFTO0FBQ1QsT0FBTyxxQ0FBcUMsQ0FBQTtBQUU1QyxhQUFhO0FBQ2IsT0FBTyxNQUFNLE1BQU0sV0FBVyxDQUFBO0FBRTlCLFNBQVM7QUFDVCxPQUFPLFFBQVEsTUFBTSx1QkFBdUIsQ0FBQTtBQUU1QyxVQUFVO0FBQ1YsT0FBTyxNQUFNLE1BQU0sbUJBQW1CLENBQUE7QUFLdEMsb0JBQW9CO0FBQ3BCLGVBQWUsTUFBTSxDQUNuQixRQUFRLEVBQ1IsTUFBTSxDQUNQLENBQUMsTUFBTSxDQUFDO0lBQ1AsSUFBSSxFQUFFLFFBQVE7SUFFZCxLQUFLLEVBQUU7UUFDTCxTQUFTLEVBQUU7WUFDVCxJQUFJLEVBQUUsQ0FBQyxNQUFNLEVBQUUsTUFBTSxDQUFDO1lBQ3RCLE9BQU8sRUFBRSxDQUFDO1NBQ1g7UUFDRCxJQUFJLEVBQUUsT0FBTztRQUNiLEtBQUssRUFBRSxPQUFPO1FBQ2QsR0FBRyxFQUFFLE1BQU07UUFDWCxNQUFNLEVBQUUsT0FBTztLQUNoQjtJQUVELFFBQVEsRUFBRTtRQUNSLE9BQU87WUFDTCxPQUFPO2dCQUNMLFFBQVEsRUFBRSxJQUFJO2dCQUNkLGVBQWUsRUFBRSxJQUFJLENBQUMsS0FBSztnQkFDM0IsR0FBRyxNQUFNLENBQUMsT0FBTyxDQUFDLFFBQVEsQ0FBQyxPQUFPLENBQUMsSUFBSSxDQUFDLElBQUksQ0FBQzthQUM5QyxDQUFBO1FBQ0gsQ0FBQztRQUNELGlCQUFpQjtZQUNmLElBQUksSUFBSSxDQUFDLElBQUk7Z0JBQUUsT0FBTyxDQUFDLENBQUE7WUFDdkIsSUFBSSxJQUFJLENBQUMsTUFBTTtnQkFBRSxPQUFPLENBQUMsQ0FBQTtZQUV6QixPQUFRLE1BQU0sQ0FBQyxPQUFPLENBQUMsUUFBZ0IsQ0FBQyxpQkFBaUIsQ0FBQyxJQUFJLENBQUMsSUFBSSxDQUFDLENBQUE7UUFDdEUsQ0FBQztRQUNELE1BQU07WUFDSixNQUFNLEtBQUssR0FBRztnQkFDWixHQUFHLE1BQU0sQ0FBQyxPQUFPLENBQUMsUUFBUSxDQUFDLE1BQU0sQ0FBQyxJQUFJLENBQUMsSUFBSSxDQUFDO2FBQzdDLENBQUE7WUFFRCxJQUFJLElBQUksQ0FBQyxHQUFHLEVBQUU7Z0JBQ1osS0FBSyxDQUFDLFVBQVUsR0FBRyxRQUFRLElBQUksQ0FBQyxHQUFHLG9DQUFvQyxDQUFBO2FBQ3hFO1lBRUQsT0FBTyxLQUFLLENBQUE7UUFDZCxDQUFDO0tBQ0Y7SUFFRCxNQUFNLENBQUUsQ0FBQztRQUNQLE1BQU0sRUFBRSxHQUFHLEVBQUUsSUFBSSxFQUFFLEdBQUcsSUFBSSxDQUFDLGlCQUFpQixDQUFDLElBQUksQ0FBQyxPQUFPLENBQUMsQ0FBQTtRQUUxRCxJQUFJLENBQUMsS0FBSyxHQUFHLElBQUksQ0FBQyxNQUFNLENBQUE7UUFFeEIsT0FBTyxDQUFDLENBQUMsR0FBRyxFQUFFLElBQUksQ0FBQyxrQkFBa0IsQ0FBQyxJQUFJLENBQUMsS0FBSyxFQUFFLElBQUksQ0FBQyxFQUFFLElBQUksQ0FBQyxNQUFNLENBQUMsT0FBTyxDQUFDLENBQUE7SUFDL0UsQ0FBQztDQUNGLENBQUMsQ0FBQSIsInNvdXJjZXNDb250ZW50IjpbIi8vIFN0eWxlc1xuaW1wb3J0ICcuLi8uLi9zdHlsdXMvY29tcG9uZW50cy9fY2FyZHMuc3R5bCdcblxuLy8gRXh0ZW5zaW9uc1xuaW1wb3J0IFZQYXBlciBmcm9tICcuLi9WUGFwZXInXG5cbi8vIE1peGluc1xuaW1wb3J0IFJvdXRhYmxlIGZyb20gJy4uLy4uL21peGlucy9yb3V0YWJsZSdcblxuLy8gSGVscGVyc1xuaW1wb3J0IG1peGlucyBmcm9tICcuLi8uLi91dGlsL21peGlucydcblxuLy8gVHlwZXNcbmltcG9ydCB7IFZOb2RlIH0gZnJvbSAndnVlJ1xuXG4vKiBAdnVlL2NvbXBvbmVudCAqL1xuZXhwb3J0IGRlZmF1bHQgbWl4aW5zKFxuICBSb3V0YWJsZSxcbiAgVlBhcGVyXG4pLmV4dGVuZCh7XG4gIG5hbWU6ICd2LWNhcmQnLFxuXG4gIHByb3BzOiB7XG4gICAgZWxldmF0aW9uOiB7XG4gICAgICB0eXBlOiBbTnVtYmVyLCBTdHJpbmddLFxuICAgICAgZGVmYXVsdDogMlxuICAgIH0sXG4gICAgZmxhdDogQm9vbGVhbixcbiAgICBob3ZlcjogQm9vbGVhbixcbiAgICBpbWc6IFN0cmluZyxcbiAgICByYWlzZWQ6IEJvb2xlYW5cbiAgfSxcblxuICBjb21wdXRlZDoge1xuICAgIGNsYXNzZXMgKCk6IG9iamVjdCB7XG4gICAgICByZXR1cm4ge1xuICAgICAgICAndi1jYXJkJzogdHJ1ZSxcbiAgICAgICAgJ3YtY2FyZC0taG92ZXInOiB0aGlzLmhvdmVyLFxuICAgICAgICAuLi5WUGFwZXIub3B0aW9ucy5jb21wdXRlZC5jbGFzc2VzLmNhbGwodGhpcylcbiAgICAgIH1cbiAgICB9LFxuICAgIGNvbXB1dGVkRWxldmF0aW9uICgpOiBudW1iZXIgfCBzdHJpbmcge1xuICAgICAgaWYgKHRoaXMuZmxhdCkgcmV0dXJuIDBcbiAgICAgIGlmICh0aGlzLnJhaXNlZCkgcmV0dXJuIDNcblxuICAgICAgcmV0dXJuIChWUGFwZXIub3B0aW9ucy5jb21wdXRlZCBhcyBhbnkpLmNvbXB1dGVkRWxldmF0aW9uLmNhbGwodGhpcylcbiAgICB9LFxuICAgIHN0eWxlcyAoKTogb2JqZWN0IHtcbiAgICAgIGNvbnN0IHN0eWxlID0ge1xuICAgICAgICAuLi5WUGFwZXIub3B0aW9ucy5jb21wdXRlZC5zdHlsZXMuY2FsbCh0aGlzKVxuICAgICAgfVxuXG4gICAgICBpZiAodGhpcy5pbWcpIHtcbiAgICAgICAgc3R5bGUuYmFja2dyb3VuZCA9IGB1cmwoXCIke3RoaXMuaW1nfVwiKSBjZW50ZXIgY2VudGVyIC8gY292ZXIgbm8tcmVwZWF0YFxuICAgICAgfVxuXG4gICAgICByZXR1cm4gc3R5bGVcbiAgICB9XG4gIH0sXG5cbiAgcmVuZGVyIChoKTogVk5vZGUge1xuICAgIGNvbnN0IHsgdGFnLCBkYXRhIH0gPSB0aGlzLmdlbmVyYXRlUm91dGVMaW5rKHRoaXMuY2xhc3NlcylcblxuICAgIGRhdGEuc3R5bGUgPSB0aGlzLnN0eWxlc1xuXG4gICAgcmV0dXJuIGgodGFnLCB0aGlzLnNldEJhY2tncm91bmRDb2xvcih0aGlzLmNvbG9yLCBkYXRhKSwgdGhpcy4kc2xvdHMuZGVmYXVsdClcbiAgfVxufSlcbiJdfQ==