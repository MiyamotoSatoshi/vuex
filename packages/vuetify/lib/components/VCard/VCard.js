var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

// Styles
import '../../../src/stylus/components/_cards.styl';
// Extensions
import VPaper from '../VPaper';
// Mixins
import Routable from '../../mixins/routable';
// Helpers
import mixins from '../../util/mixins';
/* @vue/component */
export default mixins(Routable, VPaper).extend({
    name: 'v-card',
    props: {
        elevation: {
            type: [Number, String],
            default: 2
        },
        flat: Boolean,
        hover: Boolean,
        img: String,
        raised: Boolean
    },
    computed: {
        classes: function classes() {
            return _extends({
                'v-card': true,
                'v-card--hover': this.hover
            }, VPaper.options.computed.classes.call(this));
        },
        computedElevation: function computedElevation() {
            if (this.flat) return 0;
            if (this.raised) return 3;
            return VPaper.options.computed.computedElevation.call(this);
        },
        styles: function styles() {
            var style = _extends({}, VPaper.options.computed.styles.call(this));
            if (this.img) {
                style.background = 'url("' + this.img + '") center center / cover no-repeat';
            }
            return style;
        }
    },
    render: function render(h) {
        var _generateRouteLink = this.generateRouteLink(this.classes),
            tag = _generateRouteLink.tag,
            data = _generateRouteLink.data;

        data.style = this.styles;
        return h(tag, this.setBackgroundColor(this.color, data), this.$slots.default);
    }
});
//# sourceMappingURL=VCard.js.map