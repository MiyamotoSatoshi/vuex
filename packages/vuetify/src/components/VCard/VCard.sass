@import '../../styles/styles.sass'

// Theme
+theme(v-card) using ($material)
  background-color: map-get($material, 'cards')
  color: map-deep-get($material, 'text', 'primary')

  > .v-card__text
    color: map-deep-get($material, 'text', 'secondary')

  &.v-card--outlined
    border: 1px solid map-get($material, 'dividers')

// Block
.v-card
  max-width: 100%
  outline: none
  text-decoration: none
  transition-property: box-shadow, opacity
  word-break: break-word
  overflow-wrap: break-word
  +elevation(2)

  > *:first-child:not(.v-btn):not(.v-chip),
  > .v-card__progress + *:not(.v-btn):not(.v-chip)
    border-top-left-radius: inherit
    border-top-right-radius: inherit

  > *:last-child:not(.v-btn):not(.v-chip)
    border-bottom-left-radius: inherit
    border-bottom-right-radius: inherit

// Element
.v-card__progress
  top: 0
  left: 0
  right: 0
  overflow: hidden

.v-card__title
  align-items: center
  display: flex
  flex-wrap: wrap
  font-size: map-deep-get($headings, 'h5', 'size')
  font-weight: map-deep-get($headings, 'h5', 'weight')
  letter-spacing: map-deep-get($headings, 'h5', 'letter-spacing')
  line-height: map-deep-get($headings, 'h5', 'line-height')
  padding: 16px 16px 8px

  + .v-card__text
    padding-top: 0

.v-card__text
  font-size: map-deep-get($headings, 'subtitle-2', 'size')
  line-height: map-deep-get($headings, 'subtitle-2', 'line-height')
  letter-spacing: map-deep-get($headings, 'subtitle-2', 'letter-spacing')
  padding: 16px
  width: 100%

.v-card__actions
  align-items: center
  display: flex
  padding: 8px

  .v-btn.v-btn
    padding: 0 8px

    // Buttons have reduced padding in card actions per spec
    // and need to compensate the icon margins accordingly
    +ltr()
      + .v-btn
        margin-left: 8px

      .v-icon--left
        margin-left: 4px

      .v-icon--right
        margin-right: 4px

    +rtl()
      + .v-btn
        margin-right: 8px

      .v-icon--left
        margin-right: 4px

      .v-icon--right
        margin-left: 4px

// Modifier
.v-card--flat
  +elevation(0)

.v-card--hover
  cursor: pointer
  transition: box-shadow .4s cubic-bezier(.25, .8, .25, 1)

  &:hover
    +elevation(8)

.v-card--link
  cursor: pointer

  // https://github.com/vuetifyjs/vuetify/issues/3925
  .v-chip
    cursor: pointer

  &:focus:before
    opacity: 0.08

  &:before
    background: currentColor
    bottom: 0
    content: ''
    left: 0
    opacity: 0
    pointer-events: none
    position: absolute
    right: 0
    top: 0
    transition: .2s opacity map-get($transition, 'fast-in-slow-out')

.v-card--disabled
  pointer-events: none
  user-select: none

  > *:not(.v-card__progress)
    opacity: 0.6
    transition: inherit

.v-card--loading
  overflow: hidden

.v-card--outlined
  box-shadow: none

.v-card--raised
  +elevation(8)
