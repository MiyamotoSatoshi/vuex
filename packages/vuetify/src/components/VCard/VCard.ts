// Styles
import '../../stylus/components/_cards.styl'

// Extensions
import VSheet from '../VSheet'

// Mixins
import Routable from '../../mixins/routable'

// Helpers
import mixins from '../../util/mixins'

// Types
import { VNode } from 'vue'

/* @vue/component */
export default mixins(
  Routable,
  VSheet
).extend({
  name: 'v-card',

  props: {
    elevation: {
      type: [Number, String],
      default: 2
    },
    flat: Boolean,
    hover: Boolean,
    img: String,
    raised: Boolean
  },

  computed: {
    classes (): object {
      return {
        'v-card': true,
        'v-card--hover': this.hover,
        ...VSheet.options.computed.classes.call(this)
      }
    },
    computedElevation (): number | string {
      if (this.flat) return 0
      if (this.raised) return 3

      return (VSheet.options.computed as any).computedElevation.call(this)
    },
    styles (): object {
      const style = {
        ...VSheet.options.computed.styles.call(this)
      }

      if (this.img) {
        style.background = `url("${this.img}") center center / cover no-repeat`
      }

      return style
    }
  },

  render (h): VNode {
    const { tag, data } = this.generateRouteLink(this.classes)

    data.style = this.styles

    return h(tag, this.setBackgroundColor(this.color, data), this.$slots.default)
  }
})
