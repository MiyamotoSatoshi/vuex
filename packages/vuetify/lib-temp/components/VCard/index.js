import { createSimpleFunctional } from '../../util/helpers';
import VCard from './VCard';
import VCardMedia from './VCardMedia';
import VCardTitle from './VCardTitle';
import Vue from 'vue';
const VCardActions = Vue.extend(createSimpleFunctional('v-card__actions'));
const VCardText = Vue.extend(createSimpleFunctional('v-card__text'));
export { VCard, VCardMedia, VCardTitle, VCardActions, VCardText };
export default {
    $_vuetify_subcomponents: {
        VCard,
        VCardMedia,
        VCardTitle,
        VCardActions,
        VCardText
    }
};
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvY29tcG9uZW50cy9WQ2FyZC9pbmRleC50cyJdLCJuYW1lcyI6W10sIm1hcHBpbmdzIjoiQUFBQSxPQUFPLEVBQUUsc0JBQXNCLEVBQUUsTUFBTSxvQkFBb0IsQ0FBQTtBQUMzRCxPQUFPLEtBQUssTUFBTSxTQUFTLENBQUE7QUFDM0IsT0FBTyxVQUFVLE1BQU0sY0FBYyxDQUFBO0FBQ3JDLE9BQU8sVUFBVSxNQUFNLGNBQWMsQ0FBQTtBQUNyQyxPQUFPLEdBQUcsTUFBTSxLQUFLLENBQUE7QUFFckIsTUFBTSxZQUFZLEdBQUcsR0FBRyxDQUFDLE1BQU0sQ0FBQyxzQkFBc0IsQ0FBQyxpQkFBaUIsQ0FBQyxDQUFDLENBQUE7QUFDMUUsTUFBTSxTQUFTLEdBQUcsR0FBRyxDQUFDLE1BQU0sQ0FBQyxzQkFBc0IsQ0FBQyxjQUFjLENBQUMsQ0FBQyxDQUFBO0FBRXBFLE9BQU8sRUFBRSxLQUFLLEVBQUUsVUFBVSxFQUFFLFVBQVUsRUFBRSxZQUFZLEVBQUUsU0FBUyxFQUFFLENBQUE7QUFFakUsZUFBZTtJQUNiLHVCQUF1QixFQUFFO1FBQ3ZCLEtBQUs7UUFDTCxVQUFVO1FBQ1YsVUFBVTtRQUNWLFlBQVk7UUFDWixTQUFTO0tBQ1Y7Q0FDRixDQUFBIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0IHsgY3JlYXRlU2ltcGxlRnVuY3Rpb25hbCB9IGZyb20gJy4uLy4uL3V0aWwvaGVscGVycydcbmltcG9ydCBWQ2FyZCBmcm9tICcuL1ZDYXJkJ1xuaW1wb3J0IFZDYXJkTWVkaWEgZnJvbSAnLi9WQ2FyZE1lZGlhJ1xuaW1wb3J0IFZDYXJkVGl0bGUgZnJvbSAnLi9WQ2FyZFRpdGxlJ1xuaW1wb3J0IFZ1ZSBmcm9tICd2dWUnXG5cbmNvbnN0IFZDYXJkQWN0aW9ucyA9IFZ1ZS5leHRlbmQoY3JlYXRlU2ltcGxlRnVuY3Rpb25hbCgndi1jYXJkX19hY3Rpb25zJykpXG5jb25zdCBWQ2FyZFRleHQgPSBWdWUuZXh0ZW5kKGNyZWF0ZVNpbXBsZUZ1bmN0aW9uYWwoJ3YtY2FyZF9fdGV4dCcpKVxuXG5leHBvcnQgeyBWQ2FyZCwgVkNhcmRNZWRpYSwgVkNhcmRUaXRsZSwgVkNhcmRBY3Rpb25zLCBWQ2FyZFRleHQgfVxuXG5leHBvcnQgZGVmYXVsdCB7XG4gICRfdnVldGlmeV9zdWJjb21wb25lbnRzOiB7XG4gICAgVkNhcmQsXG4gICAgVkNhcmRNZWRpYSxcbiAgICBWQ2FyZFRpdGxlLFxuICAgIFZDYXJkQWN0aW9ucyxcbiAgICBWQ2FyZFRleHRcbiAgfVxufVxuIl19