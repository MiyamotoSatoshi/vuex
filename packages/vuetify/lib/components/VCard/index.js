import { createSimpleFunctional } from '../../util/helpers';
import VCard from './VCard';
import VCardMedia from './VCardMedia';
import VCardTitle from './VCardTitle';
import Vue from 'vue';
var VCardActions = Vue.extend(createSimpleFunctional('v-card__actions'));
var VCardText = Vue.extend(createSimpleFunctional('v-card__text'));
export { VCard, VCardMedia, VCardTitle, VCardActions, VCardText };
export default {
    $_vuetify_subcomponents: {
        VCard: VCard,
        VCardMedia: VCardMedia,
        VCardTitle: VCardTitle,
        VCardActions: VCardActions,
        VCardText: VCardText
    }
};
//# sourceMappingURL=index.js.map