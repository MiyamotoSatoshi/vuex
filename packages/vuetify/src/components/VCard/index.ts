import { createSimpleFunctional } from '../../util/helpers'
import VCard from './VCard'

const VCardActions = createSimpleFunctional('v-card__actions')
const VCardText = createSimpleFunctional('v-card__text')
const VCardTitle = createSimpleFunctional('v-card__title')

export { VCard, VCardTitle, VCardActions, VCardText }

export default {
  $_vuetify_subcomponents: {
    VCard,
    VCardTitle,
    VCardActions,
    VCardText,
  },
}
