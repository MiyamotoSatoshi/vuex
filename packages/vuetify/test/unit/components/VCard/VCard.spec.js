import { test } from '@/test'
import VCard from '@/components/VCard/VCard'

test('VCard.vue', ({ mount }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(VCard)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render card with img', () => {
    const wrapper = mount(VCard, {
      propsData: {
        img: 'image.jpg'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a flat card', () => {
    const wrapper = mount(VCard, {
      propsData: {
        flat: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a raised card', () => {
    const wrapper = mount(VCard, {
      propsData: {
        raised: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render a card with custom height', () => {
    const heightpx = '400px'
    const wrapper = mount(VCard, {
      propsData: {
        height: heightpx
      }
    })

    expect(wrapper.hasStyle('height', heightpx)).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()

    wrapper.setProps({
      height: 401
    })
    expect(wrapper.hasStyle('height', '401px')).toBe(true)
  })
})
