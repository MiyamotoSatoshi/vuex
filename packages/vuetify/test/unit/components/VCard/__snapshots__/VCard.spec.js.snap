// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VCard.vue should render a card with custom height 1`] = `

<div class="v-card v-sheet theme--light"
     style="height: 400px;"
>
</div>

`;

exports[`VCard.vue should render a flat card 1`] = `

<div class="v-card v-card--flat v-sheet theme--light">
</div>

`;

exports[`VCard.vue should render a raised card 1`] = `

<div class="v-card v-sheet theme--light">
</div>

`;

exports[`VCard.vue should render card with img 1`] = `

<div class="v-card v-sheet theme--light">
</div>

`;

exports[`VCard.vue should render component and match snapshot 1`] = `

<div class="v-card v-sheet theme--light">
</div>

`;
